pkg/providers/aws/cluster_vpc.go

```diff
@@ -457,37 +457,53 @@ func getSubnets(ec2Svc ec2iface.EC2API) ([]*ec2.Subnet, error) {
 	return subs, nil
 }
 
+func getVPCIDByClusterSubnets(ec2Svc ec2iface.EC2API, clusterID string) (*string, error) {
+	var subs []*ec2.Subnet
+	listOutput, err := ec2Svc.DescribeSubnets(&ec2.DescribeSubnetsInput{})
+	if err != nil {
+		return aws.String(""), err
+	}
+	subs = listOutput.Subnets
+	for _, sub := range subs {
+		for _, tag := range sub.Tags {
+			if (*tag.Value == "owned" || *tag.Value == "shared") &&
+				*tag.Key == fmt.Sprintf("%s%s", clusterOwnedTagKeyPrefix, clusterID) {
+				return sub.VpcId, nil
+			}
+		}
+	}
+	return aws.String(""), errorUtil.New(fmt.Sprintf("failed to get cluster vpc, no vpc found: error getting vpc id from associated subnets: could not find cluster associated subnets with clusterID %s", clusterID))
```

Comment:
NP, the error is verbose, and makes sense now as the problem is fresh in our heads. Maybe changing it to something like `failed to find cluster vpc id, no subnets found with osd cluster tag`

Reply:
addressed. Rewritten error message